File: 1444-number-of-steps-to-reduce-a-number-to-zero/1444-number-of-steps-to-reduce-a-number-to-zero.cpp
class Solution {
public:
    int numberOfSteps(int num) {
        if(num==0){
            return 0;
        }
        int c=0;
        while(num>0){
            if(num%2==0){
                num/=2;
                c++;
            }
            else{
                num--;
                c+=1;
            }
        }
        return c;
        
    }
};

File: 1444-number-of-steps-to-reduce-a-number-to-zero/1444-number-of-steps-to-reduce-a-number-to-zero.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        if(num==0):
            return 0
        c=0
        while(num>0):
            if(num%2==0):
                num//=2
                c+=1
            else:
                num-=1
                c+=1
        return c
        